### DISCUSSION TITLE
Peacock won't run. It says aborted about 2 minutes after I run it.

### DISCUSSION THREAD

**Discussion by GraphicsMonster** [idaholab/moose#25580] [Q&A Tools]
I installed moose pretty recently and for visualization would like to use peacock as that's what the rest of the team is using. 
I set things up, added the path in .bashrc, set up xlaunch, initiated Xlaunch before opening the terminal and when I navigate to the exmaples directory in moose and run moose for the example I built it returns a simple "Aborted" message and no more information.

Would appreciated any sort of help. Thanks a bunch!
![peacock not working](https://github.com/idaholab/moose/assets/63590510/7073f8e1-19be-4330-8369-c0f6c2f0148a)

**Comment by milljm** [comment]
Please do not post images of text. Using images prevents any sort of search for the text contained within the image.

Are you able to display any sort of X functionality? `xeyes` for example?

**Reply by GraphicsMonster** [reply]
okay I think I got it to work. I had set the IP wrong.
But now I get an error that says segmentation fault. I am trying to run the first example in the examples folder by the way. It's called ex01_inputfile.

Can you help me with this?

**Reply by cticenhour** [reply]
If you run the input file on the command line manually, does it still segfault? 

```
cd ~/moose/examples/ex01_inputfile
./ex01-opt -i ex01.i
```

**Reply by milljm** [reply]
Interesting, I must not have hit "reply". I had something all typed up... but I do not see my post!



Do you mean to say that peacock launches correctly? But fails to run the problem?

Whats happens when you call peacock directly?
```bash
cd ~/moose/examples/ex01_inputfile
../../python/peacock/peacock -i ex01.i
```

Also, lets try to run a render test (tests most Peacock VTK/Qt/PyQt functionality):
```bash
export PYTHONPATH=$HOME/moose/python
cd ~/moose/python/chigger/tests/writers
./writers.py
```

**Reply by GraphicsMonster** [reply]
Calling peacock through path results in the same segment fault.

Peacock launches correctly and I even see a new peacock window open that says looking for executable and then the terminal says found executable with a path to it and then suddenly it shows segmentation fault and the entire operation drops right there.

Running ./writers.py gives:

`(moose) devansh@HP_15s:~/moose/python/chigger/tests/writers$ ./writers.py
Traceback (most recent call last):
  File "/home/devansh/moose/python/chigger/tests/writers/./writers.py", line 15, in <module>
    import chigger
ModuleNotFoundError: No module named 'chigger'`

Do I need to manually install this library? Is this holding it back?

**Reply by cticenhour** [reply]
Based on that output you need to make sure the MOOSE Python modules are available, as @milljm mentioned in his suggestion:

```
export PYTHONPATH=$PYTHONPATH:$HOME/moose/python
```

`chigger` is supplied within that directory.

**Reply by milljm** [reply]
Hmm interesting. I think Peacock at that moment (when its up and now searching for an executable) will have something to do with Peacock finding the executable, and something going wrong when the executable is run with the json dump args.

Can you perform the following, and report back if it completes, or dies (it will print _a lot_ of information to the screen)?

```bash
cd ~/moose/examples/ex01_inputfile
./ex01-opt --json
```

**Reply by GraphicsMonster** [reply]
This does complete. Ends with **End JSON Data**

**Comment by GraphicsMonster** [comment]
> If you run the input file on the command line manually, does it still segfault?
> 
> ```
> cd ~/moose/examples/ex01_inputfile
> ./ex01-opt -i ex01.i
> ```

No it works manually. Gives me a "solve converged" message.
No luck with peacock though.

**Reply by milljm** [reply]
I am at home using my WSL machine, and can confirm the abort message...

I have a temporary workaround, but it requires you to maintain two Conda environments. Can you try the following:

```bash
mamba deactivate
mamba create -n peacock moose-peacock=2022.12.05
mamba activate peacock
<try to launch peacock as you have been doing>
```

The issue, will be that when you want to build MOOSE or MOOSE examples, etc, you will need to operate in the 'moose' environment. You can have as many activate environments as you want, so long as you run them from different terminal windows.

**Reply by milljm** [reply]
I have created an issue to remedy this in a future `moose-dev` update: https://github.com/idaholab/moose/issues/25598

**Reply by GraphicsMonster** [reply]
Thanks a lot man! This community is super helpful.

Ran into another issue though which I can't seem to figure out. Running peacock the output says it lacks some kind of authorization and aborts. I made sure to launch vcxsrv before launching the terminal and rebuilt it using make. Am I doing something wrong? Does this work on your machine?

```
(peacock) devansh@HP_15s:~/moose/examples/ex01_inputfile$ peacock
Authorization required, but no authorization protocol specified
Aborted
(peacock) devansh@HP_15s:~/moose/examples/ex01_inputfile$ 
```

**Reply by milljm** [reply]
I was getting that... I needed to hit the little check box when launching VcXsrv, to... Oh I forget the terminology, something about access control. Its the last checkbox. I had to check it... Then restart Ubuntu WSL, then it started to work.

EDIT: I know I posted a bunch of screen shots about how I got it all working a year or so ago... I'll link the discussion when I find it!

Found it: https://github.com/idaholab/moose/discussions/22718#discussioncomment-4276193

You have to "Disable Access Control"

**Reply by GraphicsMonster** [reply]
Super sorry I know I am bothering you too much at this point but the authorization issue has  been resolved and I just ran into a segmentation error:

```
base) devansh@HP_15s:~$ conda activate peacock
(peacock) devansh@HP_15s:~$ cd moose/examples/ex01_inputfile
(peacock) devansh@HP_15s:~/moose/examples/ex01_inputfile$ peacock
QStandardPaths: XDG_RUNTIME_DIR not set, defaulting to '/tmp/runtime-devansh'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
Found executable: /home/devansh/moose/examples/ex01_inputfile/ex01-opt
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 180, 180, 180, 255'
QCssParser::parseColorValue: Specified color without alpha value but alpha given: 'rgb 111, 111, 111, 255'
Segmentation fault
(peacock) devansh@HP_15s:~/moose/examples/ex01_inputfile$ 
```
Really appreciate your help!

**Reply by milljm** [reply]
I wonder if you issue is the same as the last message posted on the link I mentioned (about a segfault).

Can you delete `ex01-opt`, and re-run peacock? Just to see if it pops up?

```bash
conda activate peacock
cd moose/examples/ex01_inputfile
rm ex01-opt
peacock
```

_Rebuilding the example is quick as all it needs to do is relink:_
```bash
conda activate moose
cd moose/examples/ex01_inputfile
make
```

**Reply by milljm** [reply]
I wonder if installing this package may help:

```bash
sudo apt-get install x11-apps libglu1-mesa libgl1-mesa-glx
                                           ^^^^^^^^^ not a part of our instructions
                                                     but I performed this on my machine
```

**Reply by GraphicsMonster** [reply]
Okay interesting. When the ex01-opt file is present it doesn't pop up and gives segmentation fault but when I delete it it does pop up but obviously nothing to display. I ran make again and tried running peacock again only to run into the segmentation fault again.

**Reply by milljm** [reply]
It's acting as though it cannot parse the dump (`ex01-opt --dump`) that Peacock needs to parse. I know you've tried this once before, but what happens you attempt to do that while using the Peacock environment?

```bash
conda activate peacock
cd moose/examples/ex01_inputfile
./ex01-opt --dump
```
Edit:

actually I believe its --json that Peacock is using:

```bash
./ex01-opt --json
```

**Reply by GraphicsMonster** [reply]
Looks like it's printing a huge json file.

Dump also returns loads of text but in another format I reckon.

**Reply by milljm** [reply]
hmm. I am not sure why it would segfault. I can only guess Peacock is parsing the output incorrectly. I'll need to think about what might be happening here.

Can you try to use Peacock on a `moose_test-opt` test?

```bash
cd moose/test
make
cd tests/kernels/simple_diffusion
peacock -i simple_diffusion.i
```
_I wrote that quick, hopefully no syntax errors_

Unfortunately I am heading home for the day... I'll have to pick this back up tomorrow. I may request to stay home so I can access a Windows machine in order to troubleshoot.

**Reply by GraphicsMonster** [reply]
Hey! Just checking in. Have you been able to solve the issue? Sorry I had my exams come up last week.
I'm still seeing the same issues with no progress.
![moose segfault](https://github.com/idaholab/moose/assets/63590510/5d0421e7-9311-44c8-bb3e-e54510dd37c4)

would really appreciate any help.

**Reply by GraphicsMonster** [reply]
A little update: Only running "peacock" with no input file specified does open a window with no input files but inside the window as soon as i hit the input files option the window gets terminated.
Trying to open a specific input file still gives segfault.
![peacock window](https://github.com/idaholab/moose/assets/63590510/c7965e21-3d87-4458-a684-8ddae6d1660c)

**Reply by milljm** [reply]
Is this Windows 11? I am reading over our thread, and I believe I mentioned turning off Access Control in VcXsrv. But if this is Windows 11, using VcXsrv might actually be the issue. As soon as Peacock tries to utilize OpenGL, it freezes/crashes.

This section from my previous link: [Problems running Peacock](https://github.com/idaholab/moose/discussions/22718#discussioncomment-4276193):

>From what I am reading, Windows 11 folks will enjoy native GL support without needing to Install [VcXsrv](https://sourceforge.net/projects/vcxsrv/reviews/). Their only pre-requisite is to install their optimized video card drivers (NVIDIA or AMD). While the rest of us still on Windows 10 continue to need VcXsrv. Considering myself something of a power user, even I was confused on Microsof's literature about WSL2/native GL support (but not on Windows 10).

If you are using `VcXsrv`, perhaps we need to tinker with some of its options. I am referring to these:

![image](https://github.com/idaholab/moose/assets/6414286/6de66755-108b-4628-a56b-0bb8cfc49c75)

(specifically the Native opengl checkbox).

Or, simply uninstall VcXsrv and try to use Window's 11 native opengl support (if you are using Windows 11)
